geode-core/src/main/java/org/apache/geode/internal/cache/tier/sockets/command/Ping.java

```diff
@@ -50,11 +51,17 @@ public void cmdExecute(final Message clientMessage, final ServerConnection serve
     }
     if (clientMessage.getNumberOfParts() > 0) {
       try {
-        DistributedMember targetServer = (DistributedMember) clientMessage.getPart(0).getObject();
-        DistributedMember myID = serverConnection.getCache().getMyId();
+        InternalDistributedMember targetServer =
+            (InternalDistributedMember) clientMessage.getPart(0).getObject();
+        InternalDistributedMember myID = serverConnection.getCache().getMyId();
         if (!myID.equals(targetServer)) {
-          pingCorrectServer(clientMessage, targetServer, serverConnection);
-          writeReply(clientMessage, serverConnection);
+          if (myID.compareTo(targetServer.getMemberIdentifier(), true, false) == 0) {
```

Comment:
The issue described [here](https://issues.apache.org/jira/browse/GEODE-8004?focusedCommentId=17090468&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-17090468) still remains, you're comparing `myID` (instance of `InternalDistributedMember`) against `targetServer.getMemberIdentifier()` (instance of `MemberIdentifier`), so the comparison fails with a `ClassCastException` and the client logs the following:

```
[warn 2020/04/24 01:33:02.869 PDT <poolTimer-edgeDescript-29> tid=0x112] Pool unexpected java.lang.ClassCastException: [B cannot be cast to java.lang.Throwable connection=Pooled Connection to rs-GEM-2885-0120a0i32xlarge-hydra-client-4:20245: Connection[DESTROYED]). Server unreachable: could not connect after 1 attempts
```